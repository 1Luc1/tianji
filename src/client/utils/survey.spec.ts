import { describe, test, expect } from 'vitest';
import {
  generateSurveyExampleCurlCode,
  generateSurveyExampleSDKCode,
} from './survey';

describe('survey', () => {
  test('example sdk code', () => {
    expect(
      generateSurveyExampleSDKCode('https://example.com', {
        id: '<surveyId>',
        workspaceId: '<workspaceId>',
        name: 'Test',
        payload: {
          items: [
            {
              name: 'textField',
              label: 'Text',
              type: 'text',
            },
          ],
        },
      })
    ).matchSnapshot();
  });

  test('example curl code', () => {
    expect(
      generateSurveyExampleCurlCode('https://example.com', {
        id: '<surveyId>',
        workspaceId: '<workspaceId>',
        name: 'Test',
        payload: {
          items: [
            {
              name: 'textField',
              label: 'Text',
              type: 'text',
            },
          ],
        },
      })
    ).matchSnapshot();
  });
});
