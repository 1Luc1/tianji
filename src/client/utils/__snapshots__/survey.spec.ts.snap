// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`survey > example curl code 1`] = `
"curl -X POST https://example.com/open/workspace/<workspaceId>/survey/<surveyId>/submit \\
  -H "Content-Type: application/json" \\
  -d '{
    "payload": {
      "textField": "Text"
    }
  }'"
`;

exports[`survey > example sdk code 1`] = `
"import { submitSurvey, initOpenapiSDK } from 'tianji-client-sdk';

initOpenapiSDK('https://example.com');

/**
 * Submit Survey into Test
 *
 * @param {text} textField Text
 */
async function submitForm(textField) {
  try {
    await submitSurvey(
      '<workspaceId>',
      '<surveyId>',
      {
        textField, // Text
      }
    );
    console.log('Submit success');
  } catch (err) {
    console.error(err);
    throw new Error(err);
  }
};"
`;
